pulsar-broker/src/main/java/org/apache/pulsar/PulsarStandaloneStarter.java

```diff
@@ -67,7 +67,7 @@ public PulsarStandaloneStarter(String[] args) throws Exception {
             zkServers = this.getAdvertisedAddress();
         } else if (isBlank(config.getAdvertisedAddress())) {
             // Use advertised address as local hostname
-            config.setAdvertisedAddress(ServiceConfigurationUtils.unsafeLocalhostResolve());
+            config.setAdvertisedAddress("localhost");
```

Comment:
I feel it is better to try to add a method to `ServiceConfigurationUtils` to resolve localhost first. It is not resolved then fallback to use "localhost".